src/main.ts

```diff
@@ -482,6 +482,26 @@ export async function readPipeline(config: Config): Promise<string> {
   const filename = path.join(folderName, config.pipeline)
   core.debug(`Reading pipeline file from ${filename}`)
   let pipeline = fs.readFileSync(filename).toString()
+  const pipelineJson = JSON.parse(pipeline)
```

Comment:
This one is interesting actually. I doubt it works because pipelines as defined today on the action they do use `{{}}` for variable substitution which I would be surprised if it does not break this parsing. You probably will have to move this logic after variable substitution.